de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/X_C_BPartner_CreditLimit.java

```diff
@@ -0,0 +1,207 @@
+/** Generated Model - DO NOT CHANGE */
+package org.compiere.model;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.util.Properties;
+
+/** Generated Model for C_BPartner_CreditLimit
+ *  @author Adempiere (generated) 
+ */
+@SuppressWarnings("javadoc")
+public class X_C_BPartner_CreditLimit extends org.compiere.model.PO implements I_C_BPartner_CreditLimit, org.compiere.model.I_Persistent 
+{
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 1945448948L;
+
+    /** Standard Constructor */
+    public X_C_BPartner_CreditLimit (Properties ctx, int C_BPartner_CreditLimit_ID, String trxName)
+    {
+      super (ctx, C_BPartner_CreditLimit_ID, trxName);
+      /** if (C_BPartner_CreditLimit_ID == 0)
+        {
+			setAmount (BigDecimal.ZERO);
+			setApprovedBy_ID (0); // @CreatedBy@
+			setC_BPartner_CreditLimit_ID (0);
+			setType (null);
+        } */
+    }
+
+    /** Load Constructor */
+    public X_C_BPartner_CreditLimit (Properties ctx, ResultSet rs, String trxName)
+    {
+      super (ctx, rs, trxName);
+    }
+
+
+    /** Load Meta Data */
+    @Override
+    protected org.compiere.model.POInfo initPO (Properties ctx)
+    {
+      org.compiere.model.POInfo poi = org.compiere.model.POInfo.getPOInfo (ctx, Table_Name, get_TrxName());
+      return poi;
+    }
+
+	/** Set Betrag.
+		@param Amount 
+		Betrag in einer definierten Währung
+	  */
+	@Override
+	public void setAmount (java.math.BigDecimal Amount)
```

Comment:
i think if we have an amount, we also need a currency here..wdyt?

Reply:
Hm.. I don't think so, because the amount will be converted all the time, I think in the currency of the client. But need to check with Mark.

Reply:
I added as a virtual column.